pulsar-broker/src/main/java/org/apache/pulsar/broker/transaction/pendingack/impl/PendingAckHandleImpl.java

```diff
@@ -300,11 +402,36 @@ public PendingAckHandleImpl(PersistentSubscription persistentSubscription) {
         return completableFuture;
     }
 
+    private void addCommitTxnRequest(TxnID txnId, Map<String, Long> properties, long lowWaterMark,
+                                    CompletableFuture<Void> completableFuture) {
+        acceptQueue.add(() -> commitTxn(txnId, properties, lowWaterMark, true).thenAccept(v ->
+                completableFuture.complete(null)).exceptionally(e -> {
+            completableFuture.completeExceptionally(e);
+            return null;
+        }));
+    }
+
     @Override
     public synchronized CompletableFuture<Void> commitTxn(TxnID txnID, Map<String, Long> properties,
-                                                          long lowWaterMark) {
+                                                          long lowWaterMark, boolean isInCacheRequest) {
```

Comment:
It seems that the param `isInCacheRequest` isn't used, do we need to add this param?

Reply:
good catch, we should add the commit request `!isInCacheRequest` in to a queue.